File: VB/Form2.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace ChartPivotInteractionApproaches
    Partial Public Class Form2
        Inherits Form


        Private browsableTable_Renamed As DataTable

        Public Property BrowsableTable() As DataTable
            Get
                Return browsableTable_Renamed
            End Get
            Set(ByVal value As DataTable)
                browsableTable_Renamed = value
                dataGridView1.DataSource = value
                RecalculateActualWidth()
            End Set
        End Property

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub dataGridView1_ColumnWidthChanged(ByVal sender As Object, ByVal e As DataGridViewColumnEventArgs) Handles dataGridView1.ColumnWidthChanged
            RecalculateActualWidth()
        End Sub

        Private Sub dataGridView1_RowHeadersWidthChanged(ByVal sender As Object, ByVal e As EventArgs) Handles dataGridView1.RowHeadersWidthChanged
            RecalculateActualWidth()
        End Sub

        Private Sub RecalculateActualWidth()
            Dim totalWidth As Integer = 0

            For Each col As DataGridViewColumn In dataGridView1.Columns
                totalWidth += col.Width
            Next col

            totalWidth += dataGridView1.RowHeadersWidth

            Width = totalWidth + 60
        End Sub

        Protected Overrides Sub OnClosing(ByVal e As CancelEventArgs)
            'base.OnClosing(e);
            e.Cancel = True
            Hide()
        End Sub

    End Class
End Namespace

File: CS/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace ChartPivotInteractionApproaches
{
    public partial class Form2 : Form
    {
        private DataTable browsableTable;

        public DataTable BrowsableTable
        {
            get { return browsableTable; }
            set
            {
                browsableTable = value;
                dataGridView1.DataSource = value;
                RecalculateActualWidth();
            }
        }
        
        public Form2()
        {
            InitializeComponent();
        }

        private void dataGridView1_ColumnWidthChanged(object sender, DataGridViewColumnEventArgs e)
        {
            RecalculateActualWidth();
        }

        private void dataGridView1_RowHeadersWidthChanged(object sender, EventArgs e)
        {
            RecalculateActualWidth();
        }

        private void RecalculateActualWidth()
        {
            int totalWidth = 0;

            foreach (DataGridViewColumn col in dataGridView1.Columns)
                totalWidth += col.Width;

            totalWidth += dataGridView1.RowHeadersWidth;

            Width = totalWidth + 60;
        }

        protected override void OnClosing(CancelEventArgs e)
        {
            //base.OnClosing(e);
            e.Cancel = true;
            Hide();
        }

    }
}